exercises/triangle/description.md

```diff
@@ -2,6 +2,16 @@ The program should raise an error if the triangle cannot exist.
 
 ## Hint
 
-The sum of the lengths of any two sides of a triangle always exceeds or 
-is equal to the length of the third side, a principle known as the _triangle
+In mathematics, a degenerate case is a limiting case in which an element of a
+class of objects is qualitatively different from the rest of the class and hence
+belongs to another, usually simpler, class.
+
+1. If the sum of the lengths of any two sides of a triangle is equal to the
+   length of the third side, there exists a degenerate triangle.
+
+2. The sum of the lengths of any two sides of a triangle always exceeds or is
```

Comment:
> you are not testing that triangles with negative sides are illegal. It seems like a good idea to test for that. The Ruby, Javascript and Coffeescript tests are currently testing for it.

Ah! It would be good to send a PR for that, then. I will first say that at #311 we had decided against it because it's not specified clearly whether the classification function is responsible for normalising possibly-negative distances into absolute lengths as opposed to the caller being responsible. If you do feel strongly about this, it is of course open to discussion if we could get a PR on it.

> What do you think of this?

Personally I am thinking it looks good, so :+1: